File: src/main/java/org/informagen/echo/resource/HtmlComponent.js
// Informagen =================================================================================
// Insure that the 'Informagen' namespace exists

if (!Core.get(window, ["Informagen"])) {
    Core.set(window, ["Informagen"], {});
}


// Informagen.HtmlComponent =======================================================================

Informagen.HtmlComponent = Core.extend(Echo.Component, {

    $load : function() {
        Echo.ComponentFactory.registerType("Informagen.HtmlComponent", this);
    },

    componentType :"Informagen.HtmlComponent"
});


// Informagen.HtmlComponent.Peer ==================================================================

Informagen.HtmlComponent.Peer = Core.extend(Echo.Render.ComponentSync, {

    $load : function() {
        Echo.Render.registerPeer("Informagen.HtmlComponent", this);
    },

    renderAdd : function(update, parentElement) {
        this.divElement = document.createElement("div");
        this.divElement.id = this.component.renderId;
        parentElement.appendChild(this.divElement);
        this.renderUpdate(update);
    },

    renderUpdate : function(update) {
        this.divElement.innerHTML = this.component.render("html", "");
        Echo.Sync.Font.render(this.component.render("font"), this.divElement);
        Echo.Sync.Color.renderFB(this.component, this.divElement);
        return false;
    },

    renderDispose : function(update) {
        delete this.divElement;
    }
});

File: src/main/java/org/informagen/echo/app/HtmlComponent.java
package org.informagen.echo.app;

import nextapp.echo.app.Component;

public class HtmlComponent extends Component {

    public static final String PROPERTY_HTML = "html";

    public HtmlComponent() {
        this("");
    }

    public HtmlComponent(String html) {
        setHtml(html);
    }

    public String getHtml() {
        return (String) get(PROPERTY_HTML);
    }

    public void setHtml(String newValue) {
        set(PROPERTY_HTML, newValue);
    }
}